app/common/auth/authentication-events.run.js

```diff
@@ -42,10 +43,14 @@ function AuthenticationEvents($rootScope, $location, Authentication, Session, _,
     function doLogout(redirect) {
         $rootScope.currentUser = null;
         $rootScope.loggedin = false;
-        if (redirect) {
-            $location.url(redirect);
-        }
-        $route.reload();
+        // we don' wat to reload until after filters are correctly set with the backend default that the user
+        // would get when logged out
+        PostFilters.resetDefaults().then(function () {
```

Comment:
the main reason behind this is to make sure that we don't reload before we reset, which causes some UI weirdness.